ir/constant.cpp

```diff
@@ -45,6 +61,31 @@ expr IntConst::getTypeConstraints() const {
          getType().sizeVar().uge(min_bits);
 }
 
+expr FloatConst::getTypeConstraints() const {
+  return true;
+}
+
+pair<expr, expr> FloatConst::toSMT_cnst() const {
+  const FloatType *ft = getType().getAsFloatType();
+  if (ft->getFpType() == FpType::Double) {
+    auto v = get_if<std::bitset<64>>(&val);
+    const auto ival = v->to_ullong();
+    double dval;
+    std::memcpy(&dval, &ival, sizeof(double));
```

Comment:
it's quite odd to me why you are converting back and forth from float/doubles to bits. Can we stick with floats/doubles for now and worry about other types when they come up?

Reply:
What you suggest was my first draft of this code. Generally, it is just a tradeoff between the support of fp128/fp80 and the clearness of the code. I am Okay with both the bitset way and the "double only" way and I'll switch it back if you prefer the "double only" way.

Reply:
I would rather have a simple version for now (double), and then we can improve as we go. Right now it seems a bit overengineered, and it also makes it harder to debug since you cannot directly inspect the value.

Reply:
fixed :-)